### ISSUE TITLE
`lein new`: always fetch latest release?

### ISSUE THREAD

**Issue by vemv** [technomancy/leiningen#2604]
Hi,

if:

* I used `lein new my-template foo` once, and Lein accordingly fetched `my-template 1.0.0`; and
* I release `my-template 2.0.0`; then
* `lein new my-template foo` will keep using `my-template 1.0.0`, even after waiting more than 20 minutes (e.g. I keep in mind the Clojars CDN TTL).

I know that one can specifiy fixed versions (or snapshots) with `lein new`, but that's not the behavior I desire: intuitively, `lein new` should always grab the latest stable (rationale: if I _don't_ specify a version, then surely I want the latest stable?).

Please consider adding a default (or option) for this.

Thanks - Victor

**Comment by glts** [comment]
This is already the behaviour today, I’ve just verified it on master.

[Documentation](https://github.com/technomancy/leiningen/blob/stable/doc/TEMPLATES.md#distributing-your-template):

> In Leiningen 2.x, templates get dynamically fetched if they're not found. So for instance `lein new heroku myproject` will find the latest version of the `heroku/lein-template` project from Clojars and use that.

The implementation does [fetch the latest `RELEASE` version](https://github.com/technomancy/leiningen/blob/2.9.1/src/leiningen/new.clj#L19).

**Comment by vemv** [comment]
Hi,

thanks for the response!

> In Leiningen 2.x, templates get dynamically fetched if they're not found. 

It says _if not found_, so what I say might make sense...

I did try this empirically twice.

**Comment by technomancy** [comment]
> lein new my-template foo will keep using my-template 1.0.0, even after waiting more than 20 minutes (e.g. I keep in mind the Clojars CDN TTL).

The behavior you want will happen, it's just that the periodicity for checking for updates defaults to 24 hours instead of 20 minutes.

**Comment by vemv** [comment]
Nice, thank you. I think that will solve a good fraction of pain.

Still, it'd be weird to tell coworkers "I updated the template, but wait 24h before using it".

I mean, the default makes sense when releasing something for the outside world, but close-knit teams have a tighter feedback loop.

tldr, coud this be made configurable (at least in the context of `lein new`)?

**Comment by technomancy** [comment]
> I mean, the default makes sense when releasing something for the outside world, but close-knit teams have a tighter feedback loop.

The docstring mentions being able to provide a specific template version; does that get you what you want for a smaller team?

**Comment by vemv** [comment]
Maybe; perhaps it'd work a few times, but it seems easy to forget providing this indication.

I feel a configuration option (such as an env var) would be more abstract: you can just people to tweak their `.bashrc`s once, then use an unencumbered `lein new` forever.

**Comment by technomancy** [comment]
The frequency of looking for updates in a remote repository is controlled by the `:update` configuration key. Does it work to set this to `:always`? If so then it may make sense for the `new` task to provide a separate default for this value when it's not specified. (The 24h default is better for non-template resolution.)

I would take a patch that implements this.

**Comment by glts** [comment]
It would be interesting to see the repro steps … this works for me on master (I don’t have to wait for 24 hours).

The 24 hours waiting period applies to snapshot versions only as far as I know.

**Comment by vemv** [comment]
@glts: when you say `this works for me` you are not specifying what 'this' is. Perhaps you are trying something slightly different.

I did provide reasonably precise reproduction steps. Which I followed twice, and also a peer did, and he observed the same exact behavior.

@technomancy has given a precise diagnostic, which also makes sense when contrasting with the source code of `lein new`.

Please make sure you are not giving an arbitrary/unnecesasary burden of proof; checking this yet again is time-consuming.

**Comment by glts** [comment]
@vemv All right, here’s what I’m doing:

* Create new template project `lein new template testdeploy` in ~/tmp
* Set project version to 3.0.0 (a new version, not used before)
* Publish to Clojars `lein deploy clojars`

Now the template is published here: https://clojars.org/testdeploy/lein-template/versions/3.0.0. Now I wait 15+ minutes for the Clojars CDN, just to be sure.

* Create new project using the template `lein new testdeploy foo` in ~/tmp

The new project is created using template version 3.0.0, downloaded from Clojars.

* In ~/tmp/testdeploy, set project version to 4.0.0
* Publish to Clojars `lein deploy clojars`

Again, the template is published here: https://clojars.org/testdeploy/lein-template/versions/4.0.0. Wait 15+ minutes.

* Create new project using the template `lein new testdeploy foo2` in ~/tmp

The new project is created using template version 4.0.0, downloaded from Clojars.

So this is the behaviour I’m seeing on master (6ab82cb52b826b0943f6bb1f96f636a90aee044a).

**Comment by hiredgunhouse** [comment]
@glts it's weird it works for you cause I am experiencing the same behavior as @vemv

**Comment by technomancy** [comment]
I've updated this in bddb37c to stop caching.
